src/main/java/world/bentobox/bentobox/util/Util.java

```diff
@@ -351,4 +359,164 @@ public static boolean isPassiveEntity(Entity entity) {
+        return PaperLib.teleportAsync(entity, location, cause);
+    }
+
+    /**
+     * Gets the chunk at the target location, loading it asynchronously if needed.
+     * @param loc Location to get chunk for
+     * @return Future that completes with the chunk
+     */
+    @Nonnull
+    public static CompletableFuture<Chunk> getChunkAtAsync(@Nonnull Location loc) {
+        return getChunkAtAsync(loc.getWorld(), loc.getBlockX() >> 4, loc.getBlockZ() >> 4, true);
+    }
+
+    /**
+     * Gets the chunk at the target location, loading it asynchronously if needed.
+     * @param loc Location to get chunk for
+     * @param gen Should the chunk generate or not. Only respected on some MC versions, 1.13 for CB, 1.12 for Paper
+     * @return Future that completes with the chunk, or null if the chunk did not exists and generation was not requested.
+     */
+    @Nonnull
+    public static CompletableFuture<Chunk> getChunkAtAsync(@Nonnull Location loc, boolean gen) {
+        return getChunkAtAsync(loc.getWorld(), loc.getBlockX() >> 4, loc.getBlockZ() >> 4, gen);
+    }
+
+    /**
+     * Gets the chunk at the target location, loading it asynchronously if needed.
+     * @param world World to load chunk for
+     * @param x X coordinate of the chunk to load
+     * @param z Z coordinate of the chunk to load
+     * @return Future that completes with the chunk
+     */
+    @Nonnull
+    public static CompletableFuture<Chunk> getChunkAtAsync(@Nonnull World world, int x, int z) {
+        return getChunkAtAsync(world, x, z, true);
+    }
+
+    /**
+     * Gets the chunk at the target location, loading it asynchronously if needed.
+     * @param world World to load chunk for
+     * @param x X coordinate of the chunk to load
+     * @param z Z coordinate of the chunk to load
+     * @param gen Should the chunk generate or not. Only respected on some MC versions, 1.13 for CB, 1.12 for Paper
+     * @return Future that completes with the chunk, or null if the chunk did not exists and generation was not requested.
+     */
+    @Nonnull
+    public static CompletableFuture<Chunk> getChunkAtAsync(@Nonnull World world, int x, int z, boolean gen) {
+        return PaperLib.getChunkAtAsync(world, x, z, gen);
+    }
+
+    /**
+     * Checks if the chunk has been generated or not. Only works on Paper 1.12+ or any 1.13.1+ version
+     * @param loc Location to check if the chunk is generated
+     * @return If the chunk is generated or not
+     */
+    public static boolean isChunkGenerated(@Nonnull Location loc) {
+        return isChunkGenerated(loc.getWorld(), loc.getBlockX() >> 4, loc.getBlockZ() >> 4);
+    }
+
+    /**
+     * Checks if the chunk has been generated or not. Only works on Paper 1.12+ or any 1.13.1+ version
+     * @param world World to check for
+     * @param x X coordinate of the chunk to check
+     * @param z Z coordinate of the chunk to checl
```

Comment:
`checl` :)

Reply:
I copied this from the PaperLib itself so haha, typo. I'll fix.